allennlp/data/dataset_readers/universal_dependencies_multilang.py

```diff
@@ -0,0 +1,171 @@
+from typing import Dict, Tuple, List
+import logging
+import itertools
+from collections import defaultdict
+import numpy as np
+
+from overrides import overrides
+
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import Field, TextField, SequenceLabelField, MetadataField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
+from allennlp.data.tokenizers import Token
+from allennlp.data.dataset_readers.universal_dependencies import lazy_parse
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+def parse_file_paths(file_paths: Dict[str, str]):
+    """
+    Converts from allennlp.params.Params to a python dict.
+
+    Parameters
+    ----------
+    file_paths :  ``Dict[str, str]``, required.
+        The dictionary of identifier (e.g. "en" for English) to file path.
+
+    Returns
+    -------
+    A dictionary of identifier ("en","it" etc.) to file path.
+    """
+    return dict(file_paths)
+
+
+@DatasetReader.register("universal_dependencies_multilang")
+class UniversalDependenciesMultiLangDatasetReader(DatasetReader):
+    """
+    Wraps UniversalDependenciesDatasetReader to support multiple input files.
+    Reads consecutive cases from each input files by the given batch_size.
+
+    Notice: when using the alternate option, one should also use the ``instances_per_epoch``
+    option for the iterator. Otherwise, each epoch will loop infinitely.
+
+    Parameters
+    ----------
+    token_indexers : ``Dict[str, TokenIndexer]``, optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        The token indexers to be applied to the words TextField.
+    use_language_specific_pos : ``bool``, optional (default = False)
+        Whether to use UD POS tags, or to use the language specific POS tags
+        provided in the conllu format.
+    alternate : ``bool``, optional (default = True)
+        Whether to alternate between input files.
+    is_first_pass_for_vocab : ``bool``, optional (default = True)
+        Whether the first pass will be for generating the vocab. If true,
+        the first pass will run over the entire dataset of each file (even if alternate is on).
+    batch_size : ``int``, optional (default = 32)
+        The amount of consecutive cases to sample from each input file when alternating.
+    """
+    def __init__(self,
+                 token_indexers: Dict[str, TokenIndexer] = None,
+                 use_language_specific_pos: bool = False,
+                 lazy: bool = False,
+                 alternate: bool = True,
+                 is_first_pass_for_vocab: bool = True,
+                 batch_size: int = 32) -> None:
+        super().__init__(lazy)
+        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
+        self.use_language_specific_pos = use_language_specific_pos
```

Comment:
We prefer to mark these things as private (with a preceding underscore), as then it's implied that users shouldn't rely on the presence of these fields, so it's easier for us to modify later without breaking backwards compatibility.  This applies to everything you've saved to `self` here.